doc/security/specs/secure_boot/index.md

```diff
@@ -6,22 +6,97 @@ title: "Secure Boot"
+3. Compute the SHA2-256 digest of the selected usage constraints concatenated with the contents of BL0, as Mask ROM does for `ROM_EXT`.
+4. Check the signature from the manifest against the digest and a public code-signing key from the current Silicon Owner (RSA-3072 with the F4 exponent).
+    * Like the Silicon Creator, the Silicon Owner may have multiple public code-signing keys. How many keys exactly is configurable by the Owner.
+    * The number of retries in case verification fails might differ from Mask ROM, and can differ between `ROM_EXT` implementations.
+5. Step the key manager device in order to obtain the Silicon Creator "root key".
+    * The root key mixes secret values provisioned by the Silicon Creator at manufacturing time with information about the specific device, the device health state, and `ROM_EXT`. It will differ between devices and between `ROM_EXT` images.
+    * The actual value of the root key is locked inside the [key manager][key-manager] and hidden from software. The key manager starts with the Silicon Creator's secret values (which are also hidden from all software) and advances its state to a new value with each additional piece of information.
+    * See the [identities and root keys][identities-keys] page for more details about the intermediate, identity, and root keys.
+6. From the Silicon Creator root key, use the key manager interface to derive the Silicon Creator "identity" key (an ECDSA-P256 key used for [attestation][attestation]) and the Silicon Owner "intermediate key" (a new key manager state which can later be used to derive the Silicon Owner's "identity" key).
+    * To create the intermediate key, the key manager starts with the Silicon Creator root key from the Mask ROM stage, and software steps the key manager state by mixing in owner secrets (provisioned by the owner at the time they take ownership of the device) and certain BL0 configuration information.
+    * In between getting the Silicon Creator identity key and the Silicon Owner intermediate key, check if the boot purpose was "attestation"; if so, generate an attestation certificate for the software/hardware state, sign it with the Silicon Creator identity key, and write it to a special region of memory.
+7. Unlock flash execution, configure ePMP so that the Silicon Creator controlled regions of memory are not writable and the BL0 region is executable, and then jump to the start of BL0.
+
+Once the code has jumped into the Silicon Owner code at BL0, secure boot in its simplest form is complete.
+The Silicon Owner may choose to extend the secure boot process with multiple boot stages of their own; this will differ between device owners, while the stages described here are guaranteed by the Silicon Creator and will be shared by all OpenTitan implementations.
+If any signature verification in the above process fails, or there is any kind of unexpected error, the device will fail to boot.
```

Comment:
Sorry for this post-merge comment.  Everything here helps a *lot* with clarifying the process, it's nice and clear.

But, should we define what the "fail to boot" behavior/state is more precisely as well?  It seems like it should be consistent for any of the possible failure modes (sig mismatch, or other errors).

We may not know exactly what is yet but when we do, it should probably go in this document.

Reply:
@alphan , do you have thoughts on this one?

Reply:
Oh, I think I misunderstood @cdgori's comment. I think we have roughly two classes of reasons that can lead to shutdown:
- Any errors (from drivers, e.g. flash_ctrl reports an error, or other code, e.g. failure to find a valid ROM_EXT image in either of the slots)
- Any interrupts or exceptions (we don't expect any in ROM).

Reply:
> * Any errors (from drivers, e.g. flash_ctrl reports an error, or other code, e.g. failure to find a valid ROM_EXT image in either of the slots)
> * Any interrupts or exceptions (we don't expect any in ROM). For exceptions, please note that we have hardenings that generate exceptions on purpose when we detect something unexpected.

Just to confirm -- both of these things trigger the same boot failure procedure, right?

I think maybe the best way to add this information is to include another paragraph at the top that says something like this:

**Boot failures**. If any step of secure boot cannot be completed (e.g. a driver returns an error code, the ROM_EXT image cannot be found), or there is any kind of interrupt or exception in Mask ROM or ROM_EXT, then the device follows a specific boot-failure process: it triggers an alert, disables critical peripherals such as the key manager and flash, re-scrambles main memory, and resets itself.

WDYT?

Reply:
> Just to confirm -- both of these things trigger the same boot failure procedure, right?

Yes, they all eventually end up in `shutdown_finalize()`. There is a minor difference for interrupts & exceptions that occur very early in the boot process, i.e. before we start executing c code: they are handled by `_asm_exception_handler` which resets the chip and kills the flash. I'm not sure if you want to go into this much detail in this doc. You may want to mention this in parentheses maybe?

 > WDYT?

Your suggestion lgtm. Thanks @jadephilipoom!